utils/substitute.go

```diff
@@ -40,17 +42,35 @@ func InitReplacers(replacers ...string) []RegexReplacer {
 			}
 			expression[2] = ""
 		}
+		// last part is optional, but specifies the timing. By default, we assume it's not important
+		if exprLen == 4 && (strings.Contains("be", strings.ToLower(expression[3]))) {
+			result[i].timing = expression[3]
+		} else if exprLen == 4 && !strings.Contains("be", strings.ToLower(expression[3])) {
+			errors.Raise("Bad timing information %b, valid values are b(egin) or e(nd)", replacers[i][3])
+		}
 		result[i].regex = regexp.MustCompile(expression[1])
 		result[i].replace = expression[2]
 
 	}
 	return result
 }
 
+// FilterReplacers will return only replacers that are marked with the right timing
+func filterReplacers(replacers []RegexReplacer, timingFilter string) []RegexReplacer {
+	var acc []RegexReplacer
```

Comment:
This is not as efficient as it could be.

In go, when you plan to filter an array, you should allocate the number of items first.

```suggestion
	acc := make([]RegexReplacer, 0, len(replacers))
```